proposals/0152-precompiles.md

```diff
@@ -0,0 +1,270 @@
+---
+simd: '0152'
+title: Precompiles
+authors:
+  - Emanuele Cesena
+category: Standard
+type: Core
+status: Draft
+created: 2024-06-03
+feature:
+supersedes:
+superseded-by:
+extends:
+---
+
+## Summary
+
+Define a unified behavior for precompiles, and highlight a few minor
+changes to the existing precompiles to minimize differences.
+
+## Motivation
+
+Precompiles are special native programs designed to verify additional
+transaction signatures.
+They run without the VM and without loading any account, and they can access
+data from other instructions, within the same transaction.
+
+At the time of writing, two precompiles exist to verify Ed25519 and
+Ethereum-like Secp256k1 signatures, and another one is being proposed to
+support Secp256r1 signatures for FIDO Passkeys.
+
+Historically, the two precompiles were built at different times and
+by different people, so naturally there are some subtle differences
+in how they behave, especially in edge cases.
+
+The main goal of this document is to provide a specification for how
+a precompile should behave, remove differences and provide guidelines
+for future proposals.
+
+In addition, we highlight 3 minor changes to the existing precompiles
+that will simplify their behavior and make it easier to develop
+alternative validator clients.
+
+## Alternatives Considered
+
+Leave as is.
+
+## New Terminology
+
+n/a
+
+## Detailed Design
+
```

Comment:
```suggestion

We first propose a new specification for a generic precompile. Then we propose changes to the existing precompiles with respect to the new specification.

```
Very minor point, but it might be confusing on first read whether the "Generic Precompile" section is describing an existing precompile or the new spec. A sentence or two to clarify that this should be the new spec going forward will help with the readability I think. We can perhaps rename "Generic Precompile" to either "Generic Precompile Specification" or just "Precompile Specification", wdyt?

Reply:
Sounds good, will rephrase

Reply:
Added, and changed the title to Precompile Specification (actually removed "generic precompile" entirely).